### PR TITLE
Switch - new IA

### PR DESCRIPTION
**Pull request by chri2547**
@palindromebart, ready for content-lgtm.

image

### CODE REVIEW

File: _component_design/switches.md [hunk 1]
```diff
@@ -19,29 +18,81 @@ helix-ui-javascript: false
...
+<div class="hxRow"  markdown="1">
+
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
+
+## When to use
+
+Use a switch in the following situations:
+
+- When you want to enable to user to make a binary decision, such as turning a setting on or off. A switch does not contain a neutral option. Modifying preferences and settings is a prime use case for this component.
+- Use a switch when the user needs to modify settings or enable billing and notification preferences. If your use case is outside of these three areas, consider a [radio]({{site.baseurl}}/components/radio-buttons.html) or [checkbox]({{site.baseurl}}/components/checkboxes.html).
```
Comment (palindromebart) [inline _component_design/switches.md]: this section is simultaneously too specific and too vague...maybe try something like this:

When a user needs to make a binary decision, such as turning a setting on or off or setting a preference enabling or disabling a device or system.

If your use case is outside of these areas, consider a [radio]({{site.baseurl}}/components/radio-buttons.html) or [checkbox]({{site.baseurl}}/components/checkboxes.html) or [selector strip] (link to selector strip page)

File: _component_design/switches.md [hunk 2]
```diff
@@ -19,29 +18,81 @@ helix-ui-javascript: false
...
-{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
+
+## Best practices
+
+Consider the following best practices when you construct a switch:
+
+- Construct the switch so that any change to its state is immediately saved. Use a checkbox or radio if the user performs additional steps for changes to be effective. For example, use a radio or checkbox if the user must click a **Submit** or **Next** button to apply changes.
+- A switch changes state when the user clicks it. Do not construct a switch that changes state when the user *clicks and drags* the knob.
+- For clarity, consider including animation that shows the user the switch behavior.
```
Comment (palindromebart) [inline _component_design/switches.md]: For clarity, consider including animation that shows knob changing position.

File: _component_design/switches.md [hunk 3]
```diff
@@ -19,29 +18,81 @@ helix-ui-javascript: false
...
+- The option label is only one word and no more than 10 characters. Use words that clearly indicate the status of the option.
```
Comment (palindromebart) [inline _component_design/switches.md]: The option label can only be on/off or yes/no. For any other options, use a selector title or input description that clearly indicates the impact of both binary states on the system.

File: _component_design/switches.md [hunk 4]
```diff
@@ -19,29 +18,81 @@ helix-ui-javascript: false
...
+- When there is a long list of settings, consider adding switches to a table. This approach provides clarity and enhances the user experience.
```
Comment (palindromebart) [inline _component_design/switches.md]: add the following sentence:

Make sure the impact of both binary states on the system is clear.

File: _component_design/switches.md [hunk 5]
```diff
@@ -106,23 +193,96 @@ Best practice form design does not allow a switch to exist in an error state, so
...
 <section class="static-section" markdown="1">
 
-## Context
+<div class="hxRow"  markdown="1">
+
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
+
+### Disabled
+
+A user can't select a switch in the disabled state. In the disabled state, the switch is locked and cannot be changed.
```
Comment (palindromebart) [inline _component_design/switches.md]: kill the first sentence here. the 2nd sentence is sufficient

File: _component_design/switches.md (hunk 1 above)
Comment (chri2547) [reply _component_design/switches.md]: Done.

File: _component_design/switches.md (hunk 2 above)
Comment (chri2547) [reply _component_design/switches.md]: Done.

File: _component_design/switches.md (hunk 3 above)
Comment (chri2547) [reply _component_design/switches.md]: Done.

File: _component_design/switches.md (hunk 4 above)
Comment (chri2547) [reply _component_design/switches.md]: Done.

File: _component_design/switches.md (hunk 5 above)
Comment (chri2547) [reply _component_design/switches.md]: Done.

**Comment by cyrichardson** [comment]
@chri2547 Some suggestions here: 
[helix_Switch_cyrEdits.docx](https://github.com/rackerlabs/design-system/files/2275288/helix_Switch_cyrEdits.docx)

File: _component_design/switches.md [hunk 6]
```diff
@@ -19,29 +18,80 @@ helix-ui-javascript: false
...
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
+
+## Best practices
+
+Consider the following best practices when you construct a switch:
+
+- Construct the switch so that any change to its state is immediately saved. Use a checkbox or radio if the user performs additional steps for changes to be effective. For example, use a radio or checkbox if the user must click a **Submit** or **Next** button to apply changes.
+- A switch changes state when the user clicks it. Do not construct a switch that changes state when the user *clicks and drags* the knob.
+- For clarity, consider including animation that shows the knob changing position.
+- The option label can only be **on/off** or **yes/no**. For any other options, use a selector title or input description that clearly indicates the impact of both binary states on the system. Ensure that the impact of both binary states on the system is clear.
```
Comment (palindromebart) [inline _component_design/switches.md]: remove the last sentence here. it's redundant.

s/he option label can only be **on/off** or **yes/no**. For any other options, use a selector title or input description that clearly indicates the impact of both binary states on the system. Ensure that the impact of both binary states on the system is clear./he option label can only be **on/off** or **yes/no**. For any other options, use a selector title or input description that clearly indicates the impact of both binary states on the system.

File: _component_design/switches.md [hunk 7]
```diff
@@ -19,29 +18,80 @@ helix-ui-javascript: false
...
+- When there is a long list of settings, consider adding switches to a table. This approach provides clarity and enhances the user experience.
```
Comment (palindromebart) [inline _component_design/switches.md]: s/When there is a long list of settings, consider adding switches to a table. This approach provides clarity and enhances the user experience./When there is a long list of settings that can be turned on/off, consider putting them in a table with a switch control right aligned in each row. This improves scanability and reduces cognitive load.

Comment (chri2547) [reply _component_design/switches.md]: Done.

File: _component_design/switches.md (hunk 6 above)
Comment (chri2547) [reply _component_design/switches.md]: Done.

**Comment by palindromebart** [comment]
LGTM!

**Pull request merged**